src/pyFAI/detectors/_common.py

```diff
@@ -1083,60 +1086,71 @@ def set_mask(self, mask):
 
-    mask = property(get_mask, set_mask)
+    # Deprecated compatibility layer
+    get_mask = deprecated(mask.fget, reason="use property", since_version="2025.09")
+    set_mask = deprecated(mask.fset, reason="use property", since_version="2025.09")
+    get_mask_crc = deprecated(mask_crc.fget, reason="use property", since_version="2025.09")
+
+
+    @property
+    def maskfile(self):
+        return self._maskfile
 
-    def set_maskfile(self, maskfile):
+    @maskfile.setter
+    def maskfile(self, maskfile):
         if fabio:
             with fabio.open(maskfile) as fimg:
                 mask = numpy.ascontiguousarray(fimg.data,
-                                               dtype=numpy.int8)
-            self.set_mask(mask)
+                                                dtype=numpy.int8)
+            self.mask = mask
             self._maskfile = maskfile
         else:
             logger.error("FabIO is not available, unable to load the image to set the mask.")
 
-    def get_maskfile(self):
-        return self._maskfile
+    # Deprecated compatibility layer
+    get_maskfile = deprecated(maskfile.fget, reason="use property", since_version="2025.09")
+    set_maskfile = deprecated(maskfile.fset, reason="use property", since_version="2025.09")
 
-    maskfile = property(get_maskfile, set_maskfile)
 
-    def get_pixel1(self):
+    @property
+    def pixel1(self) -> float:
         return self._pixel1
 
-    def set_pixel1(self, value):
-        if isinstance(value, float):
-            value = value
-        elif isinstance(value, (tuple, list)):
-            value = float(value[0])
-        else:
-            value = float(value)
+    @pixel1.setter
+    def pixel1(self, value):
+        """Set the pixel size along the first dimension."""
+        value = float(value[0] if isinstance(value, (tuple, list)) else value)
         if self._pixel1:
             err = abs(value - self._pixel1) / self._pixel1
             if self.force_pixel and (err > EPSILON):
                 logger.warning("Enforcing pixel size 1 for a detector %s" %
                                self.__class__.__name__)
         self._pixel1 = value
 
-    pixel1 = property(get_pixel1, set_pixel1)
+    # deprecated compatibility layer
+    get_pixel1 = deprecated(pixel1.fget, reason="use property", since_version="2025.09")
+    set_pixel1 = deprecated(pixel1.fset, reason="use property", since_version="2025.09")
+
 
-    def get_pixel2(self):
+    @property
+    def pixel2(self) -> float:
         return self._pixel2
 
-    def set_pixel2(self, value):
-        if isinstance(value, float):
-            value = value
-        elif isinstance(value, (tuple, list)):
-            value = float(value[0])
-        else:
-            value = float(value)
+    @pixel2.setter
+    def pixel2(self, value):
+        """Set the pixel size along the second dimension."""
+        #TODO: Is this on purpose to take the first entry in tuple, list as pixel2?
```

Comment:
I don't remind the reason for accepting a list/tuple as pixel2 ... could be to set pixel1&pixel2 but it looks pretty ugly.  Maybe one should emit a warning that this a deprecated use-case that will be removed one day. Same for pixel1 probably.